### PR TITLE
Add a PreferencesBackend api

### PR DESCRIPTION
**Pull request by barjonr**
* I did not "duplicate" the `newInstance` method in `WalletPool` api. Maybe it would be better to add a second one with the two `PreferencesBackend` not to impact clients of the current code.

### CODE REVIEW

**Comment by pollastri-pierre** [comment]
Why is the CI blocked again ><

File: core/idl/preferences/preferences.djinni [hunk 1]
```diff
@@ -74,3 +76,58 @@ PreferencesEditor = interface +c {
...
+}
+
+PreferencesChange = record {
+    type: PreferencesChangeType;
+    key: binary;
+    value: binary;
+}
+
+# Interface describing the behaviour of the backend used by Preferences.
+PreferencesBackend = interface +c {
```
Comment (alekece) [inline core/idl/preferences/preferences.djinni:92]: Add `const` qualifier on all getters

File: core/idl/preferences/preferences.djinni [hunk 2]
```diff
@@ -74,3 +76,58 @@ PreferencesEditor = interface +c {
     # Clear all preferences.
     clear();
 }
+
+PreferencesChangeType = enum {
+    PUT_TYPE;
+    DELETE_TYPE;
```
Comment (alekece) [inline core/idl/preferences/preferences.djinni:82]: The type suffix is redundant with the enumeration name
```suggestion
    PUT;
    DELETE;
```

Comment (barjonr) [reply core/idl/preferences/preferences.djinni]: I'll do this

File: core/idl/preferences/preferences.djinni (hunk 1 above)
Comment (barjonr) [reply core/idl/preferences/preferences.djinni:92]: You are right

**Comment by barjonr** [comment]
> Thanks for that work ! Can you explain the rational behind the deletion of `iterate` method ?

They were not referenced anywhere in the libcore (actually, once, in a test) and were difficult to add to the djinni API as they used `std::function` parameters. I told with Pierre about it before removing them

**Comment by alekece** [comment]
> > Thanks for that work ! Can you explain the rational behind the deletion of `iterate` method ?
> 
> They were not referenced anywhere in the libcore (actually, once, in a test) and were difficult to add to the djinni API as they used `std::function` parameters. I talked with Pierre about it before removing them

Perfect. I suggest to indicate the rational - in the PR description - for a change to avoid that case

**Pull request merged**